nncf/quantization/passes.py

```diff
@@ -85,20 +65,65 @@ def remove_shapeof_subgraphs(
             shape_of_queue.extend(nncf_graph.get_next_nodes(node) + nncf_graph.get_previous_nodes(node))
 
     nncf_graph.remove_nodes_from(nodes_to_drop)
-    return nncf_graph
 
 
-def filter_constant_nodes(
+def remove_dropout_nodes_inplace(
+    nncf_graph: NNCFGraph,
+    dropout_metatypes: List[OperatorMetatype],
+) -> None:
+    """
+    Removes the Dropout nodes from the provided NNCFGraph instance and
+        connects droput previous node with dropout next nodes inplace
+        for each dropout node.
+
+    :param nncf_graph: NNCFGraph instance for the transformation.
+    :param dropout_metatypes: List of backend-specific Dropout metatypes.
+    """
+    if not dropout_metatypes:
+        return
+
+    nodes_to_drop = []
+    for node in nncf_graph.get_all_nodes():
+        if node.metatype in dropout_metatypes:
+            nodes_to_drop.append(node)
+
+            prev_nodes = nncf_graph.get_previous_nodes(node)
+            input_edges = nncf_graph.get_input_edges(node)
+            assert len(prev_nodes) == len(input_edges) == 1
+            prev_node = prev_nodes[0]
+            input_edge = input_edges[0]
+            assert not input_edge.parallel_input_port_ids
+
+            # nncf_graph.get_next_edges is not used to preserve
+            # parallel_input_port_ids
+            for output_node in nncf_graph.get_next_nodes(node):
+                output_edge = nncf_graph.get_edge(node, output_node)
+                # Connects dropout previous node with all next nodes
+                # to keep NNCFGraph connected.
+                assert input_edge.dtype == output_edge.dtype
+                assert input_edge.tensor_shape == output_edge.tensor_shape
+                nncf_graph.add_edge_between_nncf_nodes(
+                    from_node_id=prev_node.node_id,
+                    to_node_id=output_edge.to_node.node_id,
+                    tensor_shape=input_edge.tensor_shape,
+                    input_port_id=output_edge.input_port_id,
+                    output_port_id=input_edge.output_port_id,
+                    dtype=input_edge.dtype,
+                    parallel_input_port_ids=output_edge.parallel_input_port_ids,
+                )
```

Comment:
What is the difference between removing `dropout` and any other nodes?

Reply:
Do you suggest to make this function common? With metatypy as an attribute? We don't need such function yet, that's why I made a specific function for that

Reply:
I don't see any reason to implement such a method instead of using `remove_nodes_from` yet. This is why I was asking for the difference.

Reply:
In comparison with `NNCFGraph.remove_nodes_from`, this method connects previous nodes with following one to keep NNCFGraph connected

Reply:
> this method connects previous nodes with following one to keep NNCFGraph connected

Does it mean that currently the functionality of the  `remove_nodes_from` is not correct and needs to be fixed?

Reply:
I don't know purpose of the `remove_nodes_from` function, I don't know is it correct or not

Reply:
> I don't know purpose of the remove_nodes_from function, I don't know is it correct or not

Sorry, but if you're not familiar with the `remove_nodes_from`, but trying to implement your own functionality that `connects previous nodes with ...`, doesn't it mean that basic functionality does not as expected = not correct? 

Then, why do we need the separate functionality for `dropout` nodes? It's still not clear.